design-proposals/klip-8-interactive-queries.md

```diff
@@ -131,13 +131,129 @@ Additionally:
+
+To get KSQL's current behavior with a streaming query against a table, users would simply use an intuitive `WHERE` clause:
+
+```sql
+SELECT * FROM table WHERE ROWTIME >= NOW EMIT CHANGES;
+```
+
+## Limitations
+
+The following are proposed intentional limitations of the initial implementation of PIT queries:
+
+**PIT queries on streams probably do not need to be initially supported.** The semantics and use cases around these are unclear and require further analysis. However, the proposed syntax leaves open the possibility of supporting PIT queries on streams without further modifications.
+
+**CTAS/CSAS defined by PIT queries do not initially need to be supported.** For example, the following CTAS would create a table, `t0`, and initially populate it with the result of the `SELECT` query:
+
+```sql
+CREATE TABLE t0 AS SELECT x, count(*) FROM t1 GROUP BY x;
+```
+
+**PIT queries against tables created directly over a topic (leaf tables) do not need to be initially supported.** Leaf tables are not materialized until another query reads from them. As a result, a PIT query against such a table would (potentially) not return the expected result. Rather than leave open the possibility of confusing query results in this scenario, we should simply disallow PIT queries against leaf tables until we decide to support them.
+
+---
+
+## Other considered approaches
+
+The following are other approaches that have been considered for identifying PIT versus streaming queries:
+
+### **Interpret any query against a table as PIT**
+
+The table-based approach would consider any SQL query against a KSQL table as a PIT query. Any query against a stream would then be a streaming query. The primary issue with this approach is that it complicates the capability of running a streaming query on a table, which is a highly valuable use case. Also, since PIT queries would be identified somewhat implicitly, this approach may not be particularly intuitive to users. It would also make it relatively easy to write a "correct" query that does the wrong thing.
+
+### **`SELECT STREAM[ING]`**
+
+Streaming queries would be identified by a new `STREAM[ING]` keyword: `SELECT STREAM[ING]...`. Similarly to `EMIT STREAM`, `SELECT STREAM[ING]` becomes somewhat redundant in the context of CSAS:
+
+```sql
+CREATE STREAM s0 AS SELECT STREAMING * FROM s1;
+```
+
+This approach also doesn't facilitate de-emphasizing the stream abstraction should we decide to eventually make KSQL more table centric.
+
+### **`WHERE` clause based**
+
+With this approach, streaming versus PIT behavior would be identified using a query's `WHERE` clause. For example,
+
+```sql
+SELECT * FROM  rel WHERE ROWTIME >= BEGINNING()
+```
+
+One drawback of this approach is that it may become a bit cumbersome with more complex queries such as `JOINs`:
```

Comment:
I actually think this syntax would not work at all. Assume you have a table and you want to do an "snapshot" query and express a regular filter on ROWTIME. This filter predicate should not turn your query into a continuous query.

Reply:
I just see that the same argument is presented below already.